### PR TITLE
qemu: 8.2.4 -> 9.0.0

### PR DESCRIPTION
**Pull request by alyssais**
## Description of changes

- [ChangeLog](https://wiki.qemu.org/ChangeLog/9.0)
- [Planning](https://wiki.qemu.org/Planning/9.0) (ETA: 2024-04-16)

The Darwin patches have stopped applying again. (cc @willcohen)

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [x] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by risicle** [comment]
Might it be time to set `MACOSX_DEPLOYMENT_TARGET` to something >= 10.14 for qemu?

**Comment by alyssais** [comment]
I don't use or understand macOS, so I don't really have an opinion, except that something needs to be done in the next two weeks.

**Comment by risicle** [comment]
I'm told the fixes needed to make this work on darwin are currently in staging. Haven't been able to do a staging bootstrap yet to test this.

**Comment by alyssais** [comment]
staging-next has merged now, so am I understanding correctly that if I rebase this it should build on Darwin without the patches?

**Comment by alyssais** [comment]
Still not building on x86_64-darwin.  Is there something else that needs to happen?

**Comment by willcohen** [comment]
I no longer have access to an x86 mac (moved on to arm64) and so can't actually run the relevant builds. However, unless https://github.com/NixOS/nixpkgs/issues/101229 and similar issues happen, it would surprise me to not need these anymore. The patches are basically undoing 10.13+ stuff that is being added to QEMU to allow it to keep building on x86. At this point, 10.13 came out in 2017, and I think we're reaching the point where it's unreasonable to backport software to require an OS's API that has to be 7+ years old.

If only because I don't know anyone else who is invested in qemu on darwin x86 on nix, I'm not sure how to test that functionality short of me trial-and-erroring blindly by trying to fix diffs and with ofborg as my guinea pig.

**Comment by risicle** [comment]
> staging-next has merged now, so am I understanding correctly that if I rebase this it should build on Darwin without the patches?

No, I'm told it missed this staging cycle.

**Comment by risicle** [comment]
Have tried with the new `staging-next` cycle, and still no luck :(

**Comment by matdibu** [comment]
9.0 (non RC) is out now, if you can and want to pursue this.

**Comment by alyssais** [comment]
I still need somebody to tell me what to do about x86_64-darwin, unless we just mark it broken?

**Comment by willcohen** [comment]
Unless we can find someone with an x86-darwin machine who wants to rebase and test, I think broken is the best bet, but that's just me.

**Comment by alyssais** [comment]
@ofborg test cage cagebreak cloud-init-hostname cloud-init mtp os-prober phosh sway systemd-initrd-luks-fido2 tinywl virtualbox

File: pkgs/applications/virtualization/qemu/default.nix [hunk 1]
```diff
@@ -274,6 +265,7 @@ stdenv.mkDerivation (finalAttrs: {
     license = licenses.gpl2Plus;
     maintainers = with maintainers; [ eelco qyliss ];
     platforms = platforms.unix;
+    broken = stdenv.hostPlatform.system == "x86_64-darwin";
```
Comment (matdibu) [inline pkgs/applications/virtualization/qemu/default.nix:268]: could you leave a comment here letting others know it was marked "broken" out of caution and lack of hardware to use for testing when it was updated?

Comment (alyssais) [reply pkgs/applications/virtualization/qemu/default.nix:268]: It's not marked broken "out of caution", it's marked broken because it doesn't build.

Comment (matdibu) [reply pkgs/applications/virtualization/qemu/default.nix:268]: sorry, I misread the conversation, you're right.

**Comment by alyssais** [comment]
Ugh, apparently [QEMU is release-critical](https://github.com/NixOS/nixpkgs/blob/7e9344b1061854b7e3110775527d2a8526486c5e/pkgs/top-level/release.nix#L118) for x86_64-darwin, so we can't just mark it broken without blocking the channel.

@roberth @Enzime seems like you were involved in that — can you help here?

**Comment by willcohen** [comment]
If it's okay to get it removed out of being release critical, that makes the most sense to me. If it's considered release critical, then we'll need to find a volunteer from darwin maintainers who's still on x86 and doesn't mind rebasing the patches (not pinging them yet...)

**Comment by alyssais** [comment]
@NixOS/darwin-maintainers does anybody care about QEMU on x86_64-darwin?  If nobody steps up there's not much we can do apart from remove it from the tested set…

**Comment by Et7f3** [comment]
Their is an effort to port nixos test on macos so I think we should repair it. I will try to fix it.

**Comment by toonn** [comment]
I'm also looking into this but won't have time to finish until tomorrow. Feel free to beat me to it.

**Comment by wegank** [comment]
I'm also working on this.

File: pkgs/applications/virtualization/qemu/default.nix [hunk 2]
```diff
@@ -124,11 +124,9 @@ stdenv.mkDerivation (finalAttrs: {
...
     ./revert-ui-cocoa-add-clipboard-support.patch
     # Standard about panel requires AppKit and macOS 10.13+
-    (fetchpatch {
-      url = "https://gitlab.com/qemu-project/qemu/-/commit/99eb313ddbbcf73c1adcdadceba1423b691c6d05.diff";
-      sha256 = "sha256-gTRf9XENAfbFB3asYCXnw4OV4Af6VE1W56K2xpYDhgM=";
-      revert = true;
-    })
+    ./revert-ui-cocoa-use-the-standard-about-panel.patch
+    # Safe area insets require macOS 11+
+    ./revert-ui-cocoa-use-safe-area-insets.patch
```
Comment (alyssais) [inline pkgs/applications/virtualization/qemu/default.nix:129]: @wegank This naming makes it sound like we're reverting a whole commit, but it's actually just a small part of [`91aa508d027`](https://gitlab.com/qemu-project/qemu/-/commit/91aa508d027).  Could we name it something without "revert", maybe?

Comment (wegank) [reply pkgs/applications/virtualization/qemu/default.nix:129]: Does replacing "revert" with "remove" work for you?

Comment (willcohen) [reply pkgs/applications/virtualization/qemu/default.nix:129]: I like remove. Sorry for the sloppy language before.

**Comment by alyssais** [comment]
Thanks a lot @wegank!

**Comment by Et7f3** [comment]
<details>
<summary>libvirt doesn't build (with sandbox enabled) on x86_64-darwin. It doesn't build on master</summary>

```
error: builder for '/nix/store/7h0ma9nwqn2kvk2lazq6w2mrwslf65ip-python3.11-nixops-gce-unstable-2023-05-26.drv' failed with exit code 1;
       last 10 log lines:
       > Creating a wheel...
       > * Getting build dependencies for wheel...
       > * Building wheel...
       > Successfully built nixops_gcp-1.0-py3-none-any.whl
       > Finished creating a wheel...
       > Finished executing pypaBuildPhase
       > Running phase: pythonRuntimeDepsCheckHook
       > Executing pythonRuntimeDepsCheck
       > Checking runtime dependencies for nixops_gcp-1.0-py3-none-any.whl
       >   - cryptography==40.0.1 not satisfied by version 42.0.5
       For full logs, run 'nix log /nix/store/7h0ma9nwqn2kvk2lazq6w2mrwslf65ip-python3.11-nixops-gce-unstable-2023-05-26.drv'.
error: builder for '/nix/store/d3z4nx44bqzj2l2d5zga6q1p1mfbg4rc-python3.11-nixops-hetzner-unstable-2022-04-23.drv' failed with exit code 1;
       last 10 log lines:
       > Creating a wheel...
       > * Getting build dependencies for wheel...
       > * Building wheel...
       > Successfully built nixops_hetzner-1.0-py3-none-any.whl
       > Finished creating a wheel...
       > Finished executing pypaBuildPhase
       > Running phase: pythonRuntimeDepsCheckHook
       > Executing pythonRuntimeDepsCheck
       > Checking runtime dependencies for nixops_hetzner-1.0-py3-none-any.whl
       >   - typing-extensions<4.0.0,>=3.7.4 not satisfied by version 4.11.0
       For full logs, run 'nix log /nix/store/d3z4nx44bqzj2l2d5zga6q1p1mfbg4rc-python3.11-nixops-hetzner-unstable-2022-04-23.drv'.
error: builder for '/nix/store/4pay3bg9f3bjgdgbdmb8j6pjkfyffzy1-python3.11-nixops-hetznercloud-unstable-2023-02-19.drv' failed with exit code 1;
       last 10 log lines:
       > * Getting build dependencies for wheel...
       > * Building wheel...
       > Successfully built nixops_hetznercloud-0.1.3-py3-none-any.whl
       > Finished creating a wheel...
       > Finished executing pypaBuildPhase
       > Running phase: pythonRuntimeDepsCheckHook
       > Executing pythonRuntimeDepsCheck
       > Checking runtime dependencies for nixops_hetznercloud-0.1.3-py3-none-any.whl
       >   - hcloud==1.18.2 not satisfied by version 1.35.0
       >   - typing-extensions<4.0.0,>=3.7.4 not satisfied by version 4.11.0
       For full logs, run 'nix log /nix/store/4pay3bg9f3bjgdgbdmb8j6pjkfyffzy1-python3.11-nixops-hetznercloud-unstable-2023-02-19.drv'.
error: builder for '/nix/store/jcdw0nxwwn530l4gbzipin0dwz5qhpav-python3.11-qemu-0.6.1.0a1.drv' failed with exit code 1;
       last 10 log lines:
       > Traceback (most recent call last):
       >   File "/nix/store/lfsz3kkk1gbs2cq58lkyn9f8mfdijin6-python3.11-qemu-0.6.1.0a1/bin/.qom-wrapped", line 6, in <module>
       >     from qemu.utils.qom import main
       >   File "/nix/store/lfsz3kkk1gbs2cq58lkyn9f8mfdijin6-python3.11-qemu-0.6.1.0a1/lib/python3.11/site-packages/qemu/utils/qom.py", line 41, in <module>
       >     from .qom_fuse import QOMFuse
       >   File "/nix/store/lfsz3kkk1gbs2cq58lkyn9f8mfdijin6-python3.11-qemu-0.6.1.0a1/lib/python3.11/site-packages/qemu/utils/qom_fuse.py", line 48, in <module>
       >     import fuse
       >   File "/nix/store/2gkw9jgh3f533lydjsf3n1ipxwdzhglx-python3.11-fusepy-3.0.1/lib/python3.11/site-packages/fuse.py", line 115, in <module>
       >     raise EnvironmentError('Unable to find libfuse')
       > OSError: Unable to find libfuse
       For full logs, run 'nix log /nix/store/jcdw0nxwwn530l4gbzipin0dwz5qhpav-python3.11-qemu-0.6.1.0a1.drv'.
error: builder for '/nix/store/7s782bsqjapyzf0fn303x68wm5cwzjbd-diffoscope-265.drv' failed with exit code 1;
       last 10 log lines:
       >     def test_diff(differences):
       >         expected_diff = get_data("wasm_expected_diff")
       > >       actual_diff = differences[0].unified_diff
       > E       IndexError: list index out of range
       >
       > tests/comparators/test_wasm.py:50: IndexError
       > =========================== short test summary info ============================
       > FAILED tests/comparators/test_wasm.py::test_diff - IndexError: list index out of range
       > = 1 failed, 624 passed, 55 skipped, 15 deselected, 1 xfailed, 1 xpassed in 7929.02s (2:12:09) =
       > /nix/store/79qly7vi6mjpmbmgivn4dfzl69f0f1sy-stdenv-darwin/setup: line 1588: pop_var_context: head of shell_variables not a function context
       For full logs, run 'nix log /nix/store/7s782bsqjapyzf0fn303x68wm5cwzjbd-diffoscope-265.drv'.
error: builder for '/nix/store/08vypajjpzwjvnaavyixc55m6dqhb4kx-libvirt-10.0.0.drv' failed with exit code 1;
       last 10 log lines:
       >
       >
       > Ok:                 108
       > Expected Fail:      0
       > Fail:               0
       > Unexpected Pass:    0
       > Skipped:            13
       > Timeout:            1
       >
       > Full log written to /private/tmp/nix-build-libvirt-10.0.0.drv-0/source/build/meson-logs/testlog.txt
       For full logs, run 'nix log /nix/store/08vypajjpzwjvnaavyixc55m6dqhb4kx-libvirt-10.0.0.drv'.
error: 1 dependencies of derivation '/nix/store/z32kp9vrfih3dknv42yr1fhhbw15wc8i-libvirt-glib-5.0.0.drv' failed to build
error: 1 dependencies of derivation '/nix/store/53bx2n82bcaz5syn64vs5l4h8xvm8dqd-mgmt-unstable-2022-10-24.drv' failed to build
error: 1 dependencies of derivation '/nix/store/4y9sj44dfa9mi587mc5pvab6r9q76bgy-ocaml-libvirt-0.6.1.5.drv' failed to build
error: 1 dependencies of derivation '/nix/store/mhalk47sj12mvdzs3izdwzwbmjjq9ik3-python3.11-libvirt-10.0.0.drv' failed to build
error: 1 dependencies of derivation '/nix/store/pxkp9g9lzgb262j7nx67ssm5mhylkha2-ruby3.1-ruby-libvirt-0.8.2.drv' failed to build
error: 1 dependencies of derivation '/nix/store/cns1i5nmwcic7gyrnhc20ah3prcfknp1-ruby3.2-ruby-libvirt-0.8.2.drv' failed to build
error: 1 dependencies of derivation '/nix/store/sy0a9yilhpd5ajrq4rl4zjqxzkbjaslw-ruby3.3-ruby-libvirt-0.8.2.drv' failed to build
error: 1 dependencies of derivation '/nix/store/vl5cbc0ihds0cl36n081vynvr6rjf68s-python3.11-nixops-libvirtd-unstable-2023-09-01.drv' failed to build
error: 3 dependencies of derivation '/nix/store/ma6crmv3r1rp9m7016w8ncw0j7zqkkb1-virt-manager-4.1.0.drv' failed to build
error: 2 dependencies of derivation '/nix/store/92i3g6znggb6972c27h8igic7scaz4d4-virt-viewer-11.0.drv' failed to build
error: 2 dependencies of derivation '/nix/store/c1agaklfgpbc4bf9bk6y1bdizqsgqfmx-virt-viewer-11.0.drv' failed to build
error: 1 dependencies of derivation '/nix/store/sk9skjphqjppy1k01p2gnf1mgcn14zxb-appvm-unstable-2021-12-20.drv' failed to build
error: 4 dependencies of derivation '/nix/store/mkjn9ny88zjn0v4y3i7g8s1x78hqinqs-nixops-unstable-2024-02-28.drv' failed to build
error: builder for '/nix/store/535c3f1zvw6gypijm3by9q3i9xg2bsgf-python3.12-cot-2.2.1.drv' failed with exit code 1;
       last 10 log lines:
       >     return get_versions()["version"]
       >            ^^^^^^^^^^^^^^
       >   File "/private/tmp/nix-build-python3.12-cot-2.2.1.drv-0/cot-2.2.1/versioneer.py", line 1408, in get_versions
       >     cfg = get_config_from_root(root)
       >           ^^^^^^^^^^^^^^^^^^^^^^^^^^
       >   File "/private/tmp/nix-build-python3.12-cot-2.2.1.drv-0/cot-2.2.1/versioneer.py", line 342, in get_config_from_root
       >     parser = configparser.SafeConfigParser()
       >              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
       > AttributeError: module 'configparser' has no attribute 'SafeConfigParser'. Did you mean: 'RawConfigParser'?
       > /nix/store/79qly7vi6mjpmbmgivn4dfzl69f0f1sy-stdenv-darwin/setup: line 1588: pop_var_context: head of shell_variables not a function context
       For full logs, run 'nix log /nix/store/535c3f1zvw6gypijm3by9q3i9xg2bsgf-python3.12-cot-2.2.1.drv'.
error: 16 dependencies of derivation '/nix/store/h6grs7gbcvflpknipid22d1pyafrbljs-review-shell.drv' failed to build
```

```
nix build nixpkgs#libvirt --out-link /dev/null --rebuild
error: builder for '/nix/store/fwngvfmylb17js2p7r96fkj5s2phfj87-libvirt-10.0.0.drv' failed with exit code 1;
       last 10 log lines:
       >
       >
       > Ok:                 108
       > Expected Fail:      0
       > Fail:               0
       > Unexpected Pass:    0
       > Skipped:            13
       > Timeout:            1
       >
       > Full log written to /private/tmp/nix-build-libvirt-10.0.0.drv-1/source/build/meson-logs/testlog.txt
       For full logs, run 'nix log /nix/store/fwngvfmylb17js2p7r96fkj5s2phfj87-libvirt-10.0.0.drv'.
```
</details>

Other error are not related. I think diffoscope is already broken in master (or wait for hydra)

**Comment by alyssais** [comment]
> It doesn't build on master

So the build failure is unrelated to this PR?

**Comment by reckenrode** [comment]
> libvirt doesn't build (with sandbox enabled) on x86_64-darwin. It doesn't build on master

I have found Meson-based tests to be annoyingly unreliable on Darwin. Try using `sudo`. I have no idea why that does anything, but it often works for me.

**Comment by wegank** [comment]
Result of `nixpkgs-review pr 298097` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>15 packages marked as broken and skipped:</summary>
  <ul>
    <li>mkosi-full</li>
    <li>mkosi-full.dist</li>
    <li>mkosi-full.man</li>
    <li>qemu_xen</li>
    <li>qemu_xen-light</li>
    <li>qemu_xen-light.debug</li>
    <li>qemu_xen-light.ga</li>
    <li>qemu_xen.debug</li>
    <li>qemu_xen.ga</li>
    <li>qemu_xen_4_15</li>
    <li>qemu_xen_4_15-light</li>
    <li>qemu_xen_4_15-light.debug</li>
    <li>qemu_xen_4_15-light.ga</li>
    <li>qemu_xen_4_15.debug</li>
    <li>qemu_xen_4_15.ga</li>
  </ul>
</details>
<details>
  <summary>33 packages failed to build:</summary>
  <ul>
    <li>linuxKernel.packages.linux_4_19.virtualbox</li>
    <li>linuxPackages_4_19_hardened.virtualbox (linuxKernel.packages.linux_4_19_hardened.virtualbox)</li>
    <li>linuxKernel.packages.linux_5_10.virtualbox</li>
    <li>linuxPackages_5_10_hardened.virtualbox (linuxKernel.packages.linux_5_10_hardened.virtualbox)</li>
    <li>linuxKernel.packages.linux_5_15.virtualbox</li>
    <li>linuxPackages_5_15_hardened.virtualbox (linuxKernel.packages.linux_5_15_hardened.virtualbox)</li>
    <li>linuxKernel.packages.linux_5_4.virtualbox</li>
    <li>linuxPackages_5_4_hardened.virtualbox (linuxKernel.packages.linux_5_4_hardened.virtualbox)</li>
    <li>linuxKernel.packages.linux_6_1.virtualbox</li>
    <li>linuxPackages_6_1_hardened.virtualbox (linuxKernel.packages.linux_6_1_hardened.virtualbox)</li>
    <li>linuxPackages.virtualbox (linuxKernel.packages.linux_6_6.virtualbox)</li>
    <li>linuxPackages_hardened.virtualbox (linuxPackages_6_6_hardened.virtualbox)</li>
    <li>linuxKernel.packages.linux_6_8.virtualbox</li>
    <li>linuxPackages_latest-libre.virtualbox (linuxKernel.packages.linux_latest_libre.virtualbox)</li>
    <li>linuxPackages-libre.virtualbox (linuxKernel.packages.linux_libre.virtualbox)</li>
    <li>linuxPackages_lqx.virtualbox (linuxKernel.packages.linux_lqx.virtualbox)</li>
    <li>linuxPackages_xanmod.virtualbox (linuxKernel.packages.linux_xanmod.virtualbox)</li>
    <li>linuxPackages_xanmod_latest.virtualbox (linuxKernel.packages.linux_xanmod_latest.virtualbox ,linuxPackages_xanmod_stable.virtualbox)</li>
    <li>linuxPackages_zen.virtualbox (linuxKernel.packages.linux_zen.virtualbox)</li>
    <li>linuxPackages_latest.virtualbox</li>
    <li>python312Packages.cot</li>
    <li>python312Packages.cot.dist</li>
    <li>solo5</li>
    <li>solo5.debug</li>
    <li>virtualbox</li>
    <li>virtualbox.modsrc</li>
    <li>virtualboxHardened</li>
    <li>virtualboxHardened.modsrc</li>
    <li>virtualboxHeadless</li>
    <li>virtualboxHeadless.modsrc</li>
    <li>virtualboxKvm</li>
    <li>virtualboxWithExtpack</li>
    <li>virtualboxWithExtpack.modsrc</li>
  </ul>
</details>
<details>
  <summary>56 packages built:</summary>
  <ul>
    <li>OVMFFull</li>
    <li>OVMFFull.fd</li>
    <li>alpine-make-vm-image</li>
    <li>cloud-init</li>
    <li>cloud-init.dist</li>
    <li>cloud-utils</li>
    <li>cloud-utils.guest</li>
    <li>colima</li>
    <li>cot (python311Packages.cot)</li>
    <li>cot.dist (python311Packages.cot.dist)</li>
    <li>diffoscope</li>
    <li>diffoscope.dist</li>
    <li>diffoscope.man</li>
    <li>gnome.gnome-boxes</li>
    <li>guestfs-tools</li>
    <li>libguestfs</li>
    <li>lima</li>
    <li>lima-bin</li>
    <li>lxd-lts</li>
    <li>multipass</li>
    <li>nemu</li>
    <li>open-watcom-bin</li>
    <li>open-watcom-bin-unwrapped</li>
    <li>out-of-tree</li>
    <li>python311Packages.guestfs</li>
    <li>python311Packages.guestfs.dist</li>
    <li>python311Packages.qemu</li>
    <li>python311Packages.qemu.dist</li>
    <li>python312Packages.guestfs</li>
    <li>python312Packages.guestfs.dist</li>
    <li>python312Packages.qemu</li>
    <li>python312Packages.qemu.dist</li>
    <li>qemu</li>
    <li>qemu-python-utils</li>
    <li>qemu-python-utils.dist</li>
    <li>qemu-utils</li>
    <li>qemu-utils.debug</li>
    <li>qemu.debug</li>
    <li>qemu.ga</li>
    <li>qemu_full</li>
    <li>qemu_full.debug</li>
    <li>qemu_full.ga</li>
    <li>qemu_kvm</li>
    <li>qemu_kvm.debug</li>
    <li>qemu_kvm.ga</li>
    <li>qemu_test</li>
    <li>qemu_test.debug</li>
    <li>qemu_test.ga</li>
    <li>qtemu</li>
    <li>quickemu</li>
    <li>quickgui</li>
    <li>tests.testers.nixosTest-example</li>
    <li>tests.testers.runNixOSTest-example</li>
    <li>tests.trivial-builders.references</li>
    <li>vagrant</li>
    <li>zpool-auto-expand-partitions</li>
  </ul>
</details>

**Comment by wegank** [comment]
@ofborg build solo5

**Comment by wegank** [comment]
@ofborg build virtualbox

File: pkgs/by-name/so/solo5/package.nix [hunk 3]
```diff
@@ -59,6 +59,8 @@ in stdenv.mkDerivation {
   checkPhase = ''
     runHook preCheck
     patchShebangs tests
+    substituteInPlace scripts/virtio-run/solo5-virtio-run.sh \
+      --replace-fail "-no-acpi" "-machine acpi=off"
```
Comment (alyssais) [inline pkgs/by-name/so/solo5/package.nix:63]: Upstream?

Comment (wegank) [reply pkgs/by-name/so/solo5/package.nix:63]: Solo5/solo5#579

Comment (alyssais) [reply pkgs/by-name/so/solo5/package.nix:63]: Thanks!

**Pull request merged**